test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -252,26 +280,7 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, CompositeFilterIntegrationTest,
 
 // Verifies that if we don't match the match action the request is proxied as normal, while if the
 // match action is hit we apply the specified filter to the stream.
-TEST_P(CompositeFilterIntegrationTest, TestBasic) {
-  prependCompositeFilter();
-  initialize();
-  codec_client_ = makeHttpConnection(lookupPort("http"));
-
-  {
-    auto response = codec_client_->makeRequestWithBody(default_request_headers_, 1024);
-    waitForNextUpstreamRequest();
-
-    upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
-    ASSERT_TRUE(response->waitForEndStream());
-    EXPECT_THAT(response->headers(), Http::HttpStatusIs("200"));
-  }
-
-  {
-    auto response = codec_client_->makeRequestWithBody(match_request_headers_, 1024);
-    ASSERT_TRUE(response->waitForEndStream());
-    EXPECT_THAT(response->headers(), Http::HttpStatusIs("403"));
-  }
-}
+TEST_P(CompositeFilterIntegrationTest, TestBasic) { testBasic(true); }
```

Comment:
We should also make the delegated filter `SetResponseCodeFilterConfig` a upstream filter to test this.

Alternatively you can skip `SetResponseCodeFilter` and add a new dual call-out test filter (similar to `server-factory-context-filter` in CompositeFilterSeverContextIntegrationTest) directly.  By doing this, we will have the basic integration test coverage for 3 filters together already. I guess lots of code in CompositeFilterSeverContextIntegrationTest can be refactored to be reused for this purpose. 

I think it is up to you whichever way is easier.

Reply:
Yeah, changed SetResponseCodeFilterConfig as dual filter to use it in the test.